common/hashing_test.cpp

```diff
@@ -357,33 +438,33 @@ auto AllByteStringsHashedAndSorted() {
               return static_cast<uint64_t>(lhs.hash) <
                      static_cast<uint64_t>(rhs.hash);
             });
+  CheckNoHashedDuplicates(hashes);
 
   return hashes;
 }
 
-TEST(HashingTest, Collisions1ByteSized) {
-  auto hashes_storage = AllByteStringsHashedAndSorted<1>();
-  auto hashes = llvm::ArrayRef(hashes_storage);
-
+auto ExpectNoHashCollisions(llvm::ArrayRef<HashedString> hashes) -> void {
   HashCode prev_hash = hashes[0].hash;
   llvm::StringRef prev_s = hashes[0].v;
-  for (const auto& [hash, s] : llvm::ArrayRef(hashes).slice(1)) {
+  for (const auto& [hash, s] : hashes.slice(1)) {
     if (hash != prev_hash) {
       prev_hash = hash;
       prev_s = s;
       continue;
     }
 
-    FAIL() << "Colliding hash '" << hash << "' of 1-byte strings "
+    FAIL() << "Colliding hash '" << hash << "' of strings "
```

Comment:
Can you call `FAIL` in a non-TEST function? Or should this log and `return false`, and rely on the caller to call `FAIL`?